examples/dockercoins/README.md

```diff
@@ -1,11 +1,169 @@
-# dockercoins
+# Example: Dockercoins
 
-This is the demo application originally used in Jérôme Petazzoni's [orchestration workshop](https://github.com/jpetazzo/container.training).
+In this example, we will create a Docker App where service images are built along with the App image.
 
-[Kubernetes Hands-on Workshop](https://training.play-with-kubernetes.com/kubernetes-workshop/)
+The [dockercoins](https://github.com/dockersamples/dockercoins) demo application is made up of five services:
+
+* `rng` is a web service generating random bytes
+* `hasher` is a web service computing hash of POSTed data
+* `worker` is a background process using rng and hasher
+* `webui` is the web interface to watch progress
+* `redis` is handling storage
+
+## App Definition
+
+The App definition for this example is ready to use and can be found in the [coins.dockerapp](coins.dockerapp) directory in this folder.
+
+Open the `coins.dockerapp/docker-compose.yml` file in a text editor:
+
+```yaml
+version: "3.7"
+
+services:
+  rng:
+    build: rng
+    ports:
+      - "${rng.port}:80"
+
+  hasher:
+    build: hasher
+    ports:
+      - "${hasher.port}:80"
+
+  webui:
+    build: webui
+    ports:
+      - "${webui.port}:80"
+
+  redis:
+    image: redis
+
+  worker:
+    build: worker
+```
+
+You can notice that the `rng`, `webui`, `hasher` and `worker` services all have a `build` field, i.e. each service has a Dockerfile describing how the service image must be built. 
+
+```shell
+├── coins.dockerapp
+│   ├── docker-compose.yml
+│   ├── metadata.yml
+│   └── parameters.yml
+├── hasher
+│   ├── Dockerfile
+│   └── hasher.rb
+├── rng
+│   ├── Dockerfile
+│   └── rng.py
+├── webui
+│   ├── Dockerfile
+│   ├── files
+│   │   ├── d3.min.js
+│   │   ├── index.html
+│   │   ├── jquery-1.11.3.min.js
+│   │   ├── jquery.js -> jquery-1.11.3.min.js
+│   │   ├── rickshaw.min.css
+│   │   └── rickshaw.min.js
+│   └── webui.js
+└── worker
+    ├── Dockerfile
+    └── worker.py
+```
+
+## App Image
+
+Now we are going to build an App image from this App definition. At build time, Docker App is going to build each service image then build the App image embedding the service images.
+
+```shell
+$ docker app build . -f coins.dockerapp -t myrepo/coins:0.1.0
```

Comment:
```suggestion
$ docker app build -f coins.dockerapp -t myrepo/coins:0.1.0 .
```

Just to match the documentation of `docker app build` command

Reply:
Thank you! Done.